mobile/native/src/ln_dlc/channel_status.rs

```diff
@@ -0,0 +1,197 @@
+            // We might have more than one subchannel stored, but we only care about the one that
+            // corresponds to the open LN channel.
+            match subchannels
+                .iter()
+                .find(|subchannel| subchannel.channel_id == channel_id)
+            {
+                None => ChannelStatus::LnOpen,
+                Some(subchannel) => match SubChannelState::from(subchannel) {
+                    SubChannelState::Rejected
+                    | SubChannelState::Opening
+                    | SubChannelState::CollabClosed => ChannelStatus::LnOpen,
+                    SubChannelState::Open | SubChannelState::CollabClosing => {
+                        ChannelStatus::LnDlcOpen
+                    }
+                    // We're still waiting for the LN channel to close.
+                    SubChannelState::ForceClosing | SubChannelState::ForceClosed => {
+                        ChannelStatus::LnDlcForceClosing
+                    }
+                },
+            }
+        }
+        // If the LN channel is closing or closed. To discern between the two we would need to know
+        // the status of the commitment transaction.
+        Channel {
+            channel_id: Some(channel_id),
+            channel_state:
+                ChannelState::Closed | ChannelState::ForceClosedLocal | ChannelState::ForceClosedRemote,
+            ..
+        } => {
+            match subchannels
+                .iter()
+                .find(|subchannel| subchannel.channel_id == channel_id)
+            {
+                // We never had a subchannel associated with the latest LN channel.
+                None => ChannelStatus::NotOpen,
+                Some(subchannel) => match SubChannelState::from(subchannel) {
+                    // The subchannel was never fully open.
+                    SubChannelState::Rejected | SubChannelState::Opening => ChannelStatus::NotOpen,
+                    // The subchannel was closed offchain.
+                    SubChannelState::CollabClosed => ChannelStatus::NotOpen,
+                    // The subchannel was close on-chain.
+                    SubChannelState::ForceClosed => ChannelStatus::NotOpen,
+                    // The subchannel is somehow still open even though the LN channel is not.
+                    SubChannelState::Open | SubChannelState::CollabClosing => {
+                        ChannelStatus::Inconsistent
+                    }
+                    // The subchannel is still open being closed on-chain.
+                    SubChannelState::ForceClosing => ChannelStatus::LnDlcForceClosing,
+                },
+            }
+        }
+        // If the LN channel does not have an ID associated with it, we assume that it's still not
+        // open, for it should not be usable yet.
+        Channel {
+            channel_id: None, ..
+        } => ChannelStatus::NotOpen,
+        // TODO: Do not share shadow channel model between app and coordinator.
```

Comment:
```suggestion
```

I don't think such a TODO is needed here; we have already encoded in the type system that the problematic state can't be achieved, so it's firmly in "good enough" in my book  :)

Reply:
I can move the TODO as this is just the site of a symptom, but I do think it's still relevant.

The fact that we have to handle this enum variant in the app means that we have not encoded in the type system that we can't have this variant here. Unless I'm missing something, by adding the `unreachable` we are just saying "trust us programmers, we haven't made a mistake".